InterfaceParser/parsers/rpc_base.py

```diff
@@ -0,0 +1,782 @@
+                if isinstance(param_type, EnumSubset):
+                    allowed_elements = param_type.allowed_elements
+                else:
+                    allowed_elements = param_type.elements
+                if default_value_string not in allowed_elements:
+                    raise ParseError("Default value '{}' for parameter '{}' is not a member of {} '{}'"
+                                     .format(default_value_string, params["name"], type(param_type).__name__,
+                                             params["name"]))
+                default_value = allowed_elements[default_value_string]
+            else:
+                raise ParseError("Default value specified for " + type(param_type).__name__)
+        params["default_value"] = default_value
+
+        if len(attrib) != 0:
+            raise ParseError("Unexpected attributes in parameter '{}'".format(params["name"]))
+
+        if len(subelements) != 0:
+            raise ParseError("Unexpected subelements in parameter '{}'".format(params["name"]))
+
+        return Param(**params)
+
+    def _parse_param_base_item(self, element, prefix):
+        """Parse base param items.
+
+        :param element: an instance of Element (from ElementTree)
+        :param prefix: empty string
+        :return: params, other subelements and attributes.
+        """
+        params, subelements, attrib = self._parse_base_item(element, "")
+
+        since_version = self._extract_attrib(attrib, "since")
+        if since_version is not None:
+            params["since"] = self._parse_version(since_version)
+
+        until_version = self._extract_attrib(attrib, "until")
+        if until_version is not None:
+            params["until"] = self._parse_version(until_version)
+
+        deprecated = self._extract_attrib(attrib, "deprecated")
+        if deprecated is not None:
+            params["deprecated"] = deprecated
+
+        removed = self._extract_attrib(attrib, "removed")
+        if removed is not None:
+            params["removed"] = removed
+
+        is_mandatory = self._extract_attrib(attrib, "mandatory")
+        if is_mandatory is None:
+            raise ParseError("'mandatory' is not specified for parameter '{}'".format(params["name"]))
+
+        params["is_mandatory"] = self._get_bool_from_string(is_mandatory)
+
+        scope = self._extract_attrib(attrib, "scope")
+        if scope is not None:
+            params["scope"] = scope
+
+        type_name = self._extract_attrib(attrib, "type")
+        if type_name is None:
+            raise ParseError("Type is not specified for parameter '{}'".format(params["name"]))
+        if type_name == "Boolean":
+            default_value = self._extract_attrib(attrib, "defvalue")
+            if default_value is not None:
+                default_value = self._get_bool_from_string(default_value)
+            param_type = Boolean(default_value=default_value)
+        elif type_name in ('Integer', 'Float', 'Float'):
```

Comment:
Looks like there isa  typo in here. Float is mentioned twice?

Reply:
this twice mentioning appear after renaming Double to Float.
I removed redundant mentioning.